src/components/AuthContainer.tsx

```diff
@@ -1,23 +1,40 @@
 import React, { useContext, useEffect } from 'react';
 import { SafeAreaView } from 'react-native';
 import { withAuthenticator, AmplifyTheme } from 'aws-amplify-react-native';
-import { Auth, I18n } from 'aws-amplify';
+import { Auth, I18n, API } from 'aws-amplify';
 import { Translations } from '@aws-amplify/ui-components';
 import Onboarding from './Onboarding/Onboarding';
 import AuthenticatorTheme from '../styles/AuthenticatorTheme';
 import { AppContext } from '../contexts/AppContext';
 import NavFlow from './NavContainer';
 
+import { getUser } from '../graphql/queries';
+import { UserContext } from '../contexts/UserContext';
+
 Auth.configure({ mandatorySignIn: true });
 
 function App(): JSX.Element | null {
   const { appState, setAppState } = useContext(AppContext);
+  const { userState, setUserState } = useContext(UserContext);
 
   useEffect(() => {
-    if (appState === 'Auth') {
-      setAppState('Onboarding'); // TODO: use datastore to check whether this user has at least one home
+    const getUserData = async () => {
+      const user = await Auth.currentAuthenticatedUser();
+      const result = await API.graphql({
+        query: getUser,
+        variables: { id: user.attributes.sub },
+      });
+      setUserState(result.data.getUser);
+    };
+
+    getUserData();
+
+    if (userState.homes.items.length === 0) {
+      setAppState('Onboarding');
+    } else {
+      setAppState('App');
     }
```

Comment:
Since set functions are asynchronous, I would suggest putting the if statement inside the async function and have the condition rely on the `result` instead of `userState`.
```
const getUserData = async () => {
    const user = await Auth.currentAuthenticatedUser();
    const result = await API.graphql({
        query: getUser,
        variables: { id: user.attributes.sub },
    });
    const userData = result.data.getUser;
    setUserState(userData);
    if (userData.homes.items.length === 0) {
      setAppState('Onboarding');
    } else {
      setAppState('App');
    }
};

getUserData();
```

Reply:
This is intentional since putting that if statement in the function forces multiple updates to the appState. The screen flashes between the onboarding and App pages and does not display the loading icon properly.

Reply:
Thanks for clarifying. That makes sense since you want to show loading state when the userState is not yet updated.